docs/overview.adoc

```diff
@@ -19,25 +19,141 @@ For example using Kroxylicious proxy it's easier to experiment with alternative
+|     |          |        |                 |     :               :         |
+|     .----------.        |                 |     :  .----------. :         |
+|                         |                 |     :  |[Broker2] | :         |
+|                         |                 +------->|          | :         |
+|                         |                 |     :  |          | :         |
+|                         |                 |     :  .----------. :         |
+|     .----------.        |                 |     :               :         |
+|     |[Client]  |        |                 |     :  .----------. :         |
+|     |          |        |                 |     :  |[Broker3] | :         |
+|     |          +------->|                 +------->|          | :         |
+|     |          |        |                 |     :  |          | :         |
+|     .----------.        |                 |     :  .----------. :         |
+|                         .-----------------.     .===============+         |
+|                                  :                                        |
+|                                  :                                        |
+|     [downstream]                 :                  [upstream]            |
+.---------------------------------------------------------------------------.
+[0,0]: {"fill":"#99d","a2s:delref":1}
+....
+
+Now let's define some concepts used within Kroxylicious itself.
+
+=== Virtual Cluster
+
+The _Virtual Cluster_ is the downstream representation of a Kafka Cluster.  At the conceptual level, a Kafka Client
+connects to a Virtual Cluster.  Kroxylicious proxies all communications made to the Virtual Cluster through to a
+(physical) Kafka Cluster, passing it through the _Filter Chain_.
+
+So far, this explanation has elided the detail of Kafka Brokers.  Let's talk about that now.
+
+The Virtual Cluster automatically exposes a bootstrap endpoint for the Virtual Cluster.  This is what the Kafka Client
+must specify as the https://kafka.apache.org/documentation/#producerconfigs_bootstrap.servers[bootstrap.servers] property
+in the client configuration.
+
+In addition to the bootstrap endpoint, Kroxylicious automatically exposes broker endpoints.  There is one broker endpoint
+for each broker of the physical cluster.  When the Client connects to a broker endpoint, Kroxylicious proxies all
+communications to the corresponding broker of the (physical) Kafka Cluster.
+
+Kroxylicious automatically intercepts all the Kafka RPC responses that contain a broker address.  It rewrites the address
+so that it refers to the corresponding broker endpoint of the Virtual Cluster.  This means when the Kafka Client
+goes to connect to, say broker 0, it does so through the Virtual Cluster.
+
+=== Target Cluster
+
+The _Target Cluster_ is the definition of physical Kafka Cluster within the Kroxylicious itself.
+
+A _Virtual Cluster_ has exactly one _Target Cluster_.
```

Comment:
I think supporting a virtual cluster comprising many physical clusters would be a major piece of work: `brokerIds` would need to be mapped in some stable way to prevent collisions, partition reassignments would get funky, there would be a the possibility of topic/group name clashes...  I'm also not sure of what benefit a composite cluster would bring over a just a larger Kafka Cluster.     Anyway, conversations for another place (maybe worth while starting a discussion).

I'll add a sentence to say it is out of scope.